Modules/LearningSequence/classes/Player/LSControlBuilder.php

```diff
@@ -293,4 +302,37 @@ public function getStartControl()
 		return $this->start;
 	}
 
+	protected function getStartSignal(): ILIAS\UI\Component\Signal
+	{
+		$id = uniqid();
+		$signal = new ILIAS\UI\Implementation\Component\Signal($id);
+		return $signal;
+	}
+
+
+	public function getAdditionalJS(): string
+	{
+		return $this->additional_js;
+	}
+
+	protected function setListenerJS(string $id, string $new_win_url, string $on_new_win_close_url)
+	{
+		$this->additional_js =
+<<<JS
+$(document).on('{$id}', function() {
+
+	var il_ls_win = window.open('$new_win_url'),
+		il_ls_win_watch = setInterval(
+			function(){
+				if (il_ls_win.closed) {
+					clearInterval(il_ls_win_watch);
+					location.replace('$on_new_win_close_url');
+				}
+			},
```

Comment:
What is this supposed to do?

Reply:
This opens a new window/tab and periodically checks if the new win was closed. If so, the current page should be reloaded, so that e.g. LearningProgress or state changes are visible.